test/commons/dom/is-hidden-with-css.js

```diff
@@ -0,0 +1,225 @@
+			'<main style="' +
+			mainProps +
+			'"><p style="' +
+			targetProps +
+			'"></p></main>';
+		return group;
+	}
+
+	function makeShadowTree(node, mainProps, targetProps) {
+		var root = node.attachShadow({ mode: 'open' });
+		var node = createContentSlotted(mainProps, targetProps);
+		root.appendChild(node);
+	}
+
+	afterEach(function() {
+		document.getElementById('fixture').innerHTML = '';
+	});
+
+	it('should throw an error if computedStyle returns null', function*() {
+		var orig = window.getComputedStyle;
+		var fakeNode = {
+			nodeType: Node.ELEMENT_NODE,
+			nodeName: 'div'
+		};
+		window.getComputedStyle = function() {
+			return null;
+		};
+		var actual = isHiddenWithCSSFn(fakeNode);
+		var expected = Error;
+		assert.throws(actual, expected);
+		window.getComputedStyle = orig;
+	});
+
+	it('should return false on static-positioned, visible element', function() {
+		fixture.innerHTML = '<div id="target">Hello!</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return true on static-positioned, hidden element', function() {
+		fixture.innerHTML = '<div id="target" style="display:none">Hello!</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isTrue(actual);
+	});
+
+	it('should return false on absolutely positioned elements that are on-screen', function() {
+		fixture.innerHTML =
+			'<div id="target" style="position: absolute; left: 10px; right: 10px">hi</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return false for off-screen and aria-hidden element', function() {
+		fixture.innerHTML =
+			'<button id="target" aria-hidden=“true” style=“position:absolute: top:-999em”>I am Css Visible</button>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return false on fixed position elements that are on-screen', function() {
+		fixture.innerHTML =
+			'<div id="target" style="position:fixed; bottom: 0; left: 0;">StickySticky</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return false for off-screen absolutely positioned element', function() {
+		fixture.innerHTML =
+			'<div id="target" style="position: absolute; left: -9999px">Hi</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return false for off-screen fixed positioned element', function() {
+		fixture.innerHTML =
+			'<div id="target" style="position: fixed; top: -9999px">Hi</div>';
+		var el = document.getElementById('target');
+		var actual = isHiddenWithCSSFn(el);
+		assert.isFalse(actual);
+	});
+
+	it('should return false on detached elements', function() {
```

Comment:
When would this case arise? Would these even be returned from querying the DOM?

Reply:
You are right, probably won't be returned when querying the DOM. No harm in this test case, as that is exactly what we are testing.